### PR TITLE
v1.2.0

### PR DESCRIPTION
**Pull request by jorgemasta**
# 1.2.0
### Embedded checkout

- Now the `SHOP_TOKEN` is set at TLD (Top Level Domain) so the subdomain (where the embedded checkout iframe lives) can access the token
- Added a new section in the readme with some notes about how to make the checkout works
- Modified the `logout` endpoint to be `POST` to prevent unwanted caching

*Resolves [#36](https://github.com/bigcommerce/storefront-data-hooks/issues/36)*
*Resolves [#6](https://github.com/bigcommerce/storefront-data-hooks/issues/6)*

### Create `useOrders` hooks

Allow to fetch the orders using a new `useOrders` hook.

- If the user is not logged in, nothing is returned.
- If the user is logged in, it returns an array with his orders.
- Get the customer ID on the server (through the token) to avoid security issues.

*Resolves [#25](https://github.com/bigcommerce/storefront-data-hooks/issues/25)*

### Add locale to cart

When creating the carts, you can specify the locale. By default it's **en**, now the package checks the locale of the app to add it to the request.

⚠️ This adds as a Peer Dependency `next`.

*Resolves [#30](https://github.com/bigcommerce/storefront-data-hooks/issues/30)*

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,30 @@
...
+
+*Resolves [#36](https://github.com/bigcommerce/storefront-data-hooks/issues/36)*
+*Resolves [#6](https://github.com/bigcommerce/storefront-data-hooks/issues/6)*
+
+### Create `useOrders` hooks
+
+Allow to fetch the orders using a new `useOrders` hook.
+
+- If the user is not logged in, nothing is returned.
+- If the user is logged in, it returns an array with his orders.
```
Comment (amckemie) [inline CHANGELOG.md:16]: change his to their (inclusivity :)

File: README.md [hunk 2]
```diff
@@ -391,6 +392,19 @@ const { product } = await getProduct({
...
+## Checkout
+
+> The checkout only works on **production** and with a custom domain
+
+The recommended method is the [Embedded Checkout](https://developer.bigcommerce.com/api-docs/storefronts/embedded-checkout/embedded-checkout-tutorial), follow the tutorial to create a channel and a site. Notes:
+
+- The channel should be of type `storefront`
+- The site url must be your production url (e.g: https://mystore.com)
+- This package takes care of the cart and redirect links creation
+- Your bigcommerce store must be a subdomain of your headless store (eg: https://bc.mystore.com)
```
Comment (amckemie) [inline README.md:404]: Let's change the terms "store" to "storefront", as the store is the BC instance associated with an account and storefronts the individual channels.

File: src/api/customers/logout.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ export type LogoutHandlers = {
   logout: BigcommerceHandler<null, { redirectTo?: string }>
 }
 
-const METHODS = ['GET']
+const METHODS = ['POST']
```
[the remark this answers is not in the document]
Comment (jorgemasta) [reply src/api/customers/logout.ts:14]: Nice catch, thanks!
I have added an array of (allowed but warned) `DEPRECATED_METHODS`.

[the remark this answers is not in the document]
Comment (amckemie) [reply src/api/customers/logout.ts:14]: Boom! Sometimes I get to feel smart, haha :)

File: README.md [hunk 4]
```diff
@@ -391,6 +392,19 @@ const { product } = await getProduct({
...
+## Checkout
+
+> The checkout only works on **production** and with a custom domain
+
+The recommended method is the [Embedded Checkout](https://developer.bigcommerce.com/api-docs/storefronts/embedded-checkout/embedded-checkout-tutorial), follow the tutorial to create a channel and a site. Notes:
+
+- The channel should be of type `storefront`
+- The site url must be your production url (e.g: https://mystorefront.com)
+- This package takes care of the cart and redirect links creation
+- Your bigcommerce store must be a subdomain of your headless store (eg: https://bc.mystorefront.com)
```
Comment (amckemie) [inline README.md:404]: @jorgemasta Looks like this didn't get updated to "storefront" for both usages of "store".

**Review by amckemie — APPROVED** [review]

**Comment by avisra** [comment]
@jorgemasta @amckemie Hey guys - I saw that this is utilizing the V2 orders API. For our implementation, we need to mimic the screens in the default BigCommerce theme. After tinkering with this, here is what I saw lacking:
- Images for each order (one of the product images) on list view
- Detail information loaded in (addresses, shipments, etc) on detail view
- Product (variant) images for on the detail screen

I tried to implement these features and it is **very slow**. Getting the order details alone requires like 3-5 API calls. And on the order list view, it has to:
1. Get the orders for a customer
2. For each order, get the products - get the first product in the order
3. Fetch product images for that first product

If you have 10 orders, that turns into 21 separate API calls.

If you have any pull with the API team @ BigCommerce, pleasseee convince them to deliver a GraphQL verson of the orders API (or even a V3 which gives us more control over what comes back).

**Comment by amckemie** [comment]
@avisra Thank you for the input on this, we appreciate the valuable feedback! This has been shared with the correct teams internally, but at this time we can't provide any directional timelines on this, as currently we are focusing on adding support for cart, checkout, and more catalog sorting/filtering capabilities. However, this is an item on the roadmap for graphQL.

**Pull request merged**